ui/platform_window/x11/x11_window_base.cc

```diff
@@ -534,17 +536,38 @@ void X11WindowBase::OnWMStateUpdated() {
   // Propagate the window minimization information to the client.
   ui::PlatformWindowState state =
       ui::PlatformWindowState::PLATFORM_WINDOW_STATE_NORMAL;
-  if (IsMinimized() != was_minimized) {
-    if (IsMinimized()) {
-      state = ui::PlatformWindowState::PLATFORM_WINDOW_STATE_MINIMIZED;
-    } else if (IsMaximized()) {
-      // When the window is recovered from minimized state, set state to the
-      // previous maximized state if it was like that. Otherwise, NORMAL state
-      // will be set.
-      state = ui::PlatformWindowState::PLATFORM_WINDOW_STATE_MAXIMIZED;
-    }
+  if (IsMaximized())
+    state = ui::PlatformWindowState::PLATFORM_WINDOW_STATE_MAXIMIZED;
+  else if (IsMinimized())
+    state = ui::PlatformWindowState::PLATFORM_WINDOW_STATE_MINIMIZED;
+
+  // |restored_bounds_in_pixels_| can tell if the maximize/restore has been
+  // triggered by client or not. Typically, if it was the client who triggered
+  // that, |restored_bounds_in_pixels_| had values stored. Thus, we can be
+  // sure the client is not flooded with window states when the sources of a
+  // change has been the client itself.
+  if ((restored_bounds_in_pixels_.IsEmpty() &&
+       IsMaximized() != was_maximized) ||
+      IsMinimized() != was_minimized) {
     delegate_->OnWindowStateChanged(state);
+    // TODO(msisov, tonikitoo): found a better way to force-update the layout
+    // of opaque view.
+    // A workaround to update the opaque view layout:
```

Comment:
slightly wording suggestion: 

 - remove line 556
 +    // When the state of a window changed by the host
 +    // window manager in reaction to an user action (e.g. window is moved up to the system top tray and it becomes 
 +    // automatically maximized/restored), its bounds must be sent right after
 +    // the state change. But, for instance, in gnome with either XWayland or
 +    // X11, the bounds are updated first and the state is updated
 +    // second. Bounds change makes |OpaqueBrowserFrameViewLayout| to relayout
 +    // but it fails to track the right window state as long as it has not been
 +    // updated before the call to |OnWindowStateChanged|.

Reply:
This is completely removed and not needed any more.